odl/space/space_utils.py

```diff
@@ -93,168 +105,492 @@ def vector(array, dtype=None, impl='numpy'):
+
+    Returns
+    -------
+    real_tspace : `BaseTensorSpace`
+
+    Examples
+    --------
+    Space of real 2x3 tensors with ``float32`` entries:
+
+    >>> odl.rtensors((2, 3), dtype='float32')
+    rtensors((2, 3), 'float32')
+
+    The default data type depends on the implementation. For
+    ``impl='numpy'``, it is ``'float64'``:
 
-        Only complex floating-point data types are allowed.
+    >>> ts = odl.rtensors((2, 3))
+    >>> ts
+    rtensors((2, 3))
+    >>> ts.dtype
+    dtype('float64')
 
-        Default: default of the implementation given by calling
-        ``default_dtype(ComplexNumbers())`` on the `FnBase` implementation.
+    One-dimensional spaces have special constructors:
+
+    >>> odl.rtensors((3,))
+    rn(3)
+
+    See also
+    --------
+    tensor_space : Space of tensors with arbitrary scalar data type.
+    ctensors : Complex tensor space.
+    """
+    real_tspace_type = TENSOR_SPACE_IMPLS[impl]
+
+    if dtype is None:
+        dtype = real_tspace_type.default_dtype(RealNumbers())
+
+    real_tspace = real_tspace_type(shape, dtype, order, **kwargs)
+
+    if not real_tspace.is_real_space:
+        raise ValueError('data type {!r} not a real floating-point type'
+                         ''.format(dtype))
+
+    return real_tspace
+
+
+def vector(array, dtype=None, impl='numpy'):
+    """Create an n-tuples type vector from an array-like object.
+
+    Parameters
+    ----------
+    array : `array-like`
+        Array from which to create the vector. Scalars become
+        one-dimensional vectors.
+    dtype : optional
+        Set the data type of the vector manually with this option.
+        By default, the space type is inferred from the input data.
     impl : string
-        The backend to use. See `odl.space.entry_points.FN_IMPLS` for
-        available options.
-    kwargs :
-        Extra keyword arguments to pass to the implmentation.
+        The backend to use. See `odl.space.entry_points.TENSOR_SET_IMPLS`
+        and `odl.space.entry_points.TENSOR_SPACE_IMPLS` for available
+        options.
 
     Returns
     -------
-    cn : `FnBase`
+    vec : `BaseGeneralizedTensor`
+        Vector created from the input array. Its concrete type depends
+        on the provided arguments.
+
+    Notes
+    -----
+    This is a convenience function and not intended for use in
+    speed-critical algorithms.
+
+    Examples
+    --------
+    >>> odl.vector([1, 2, 3])  # No automatic cast to float
+    fn(3, 'int').element(
+    [1, 2, 3]
+    )
+    >>> odl.vector([1, 2, 3], dtype=float)
+    rn(3).element(
+    [1.0, 2.0, 3.0]
+    )
+    >>> odl.vector([1 + 1j, 2, 3 - 2j])
+    cn(3).element(
+    [(1+1j), (2+0j), (3-2j)]
+    )
+
+    Non-scalar types are also supported:
+
+    >>> odl.vector([True, False])
+    ntuples(2, 'bool').element(
+    [True, False]
+    )
+
+    Scalars become a one-element vector:
+
+    >>> odl.vector(0.0)
+    rn(1).element(
+    [0.0]
```

Comment:
Reading this makes me even more strongly convinced that we should AT LEAST indent the number here, but likely try to fit it on one line if possible.